### PR TITLE
[1.11.x] [RHPAM-4053] - Kogito builder image: Source file generation should us…

### PR DESCRIPTION
**Pull request by spolti**
…e productized maven quarkus plugin (different GAV) and specify the quarkus platform id/version
See: https://issues.redhat.com/browse/RHPAM-4053

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
seems not to be related with the changes made.
@radtriste maybe it is related with the graalvm version?
```
[error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateModel threw an exception: java.lang.IllegalStateException: src/main/java/org/kie/kogito/app/GlobalObjectMapper.java (24:914) : Only a type can be imported. io.quarkus.jackson.ObjectMapperCustomizer resolves to a package
[ERROR] src/main/java/org/kie/kogito/app/GlobalObjectMapper.java (27:1012) : ObjectMapperCustomizer cannot be resolved to a type
[ERROR] src/main/java/org/kie/kogito/app/GlobalObjectMapper.java (24:914) : Only a type can be imported. io.quarkus.jackson.ObjectMapperCustomizer resolves to a packagesrc/main/java/org/kie/kogito/app/GlobalObjectMapper.java (27:1012) : ObjectMapperCustomizer cannot be resolved to a type
```

**Review by sutaakar — APPROVED** [review]

**Comment by radtriste** [comment]
@spolti what about `-DplatformGroupId`, `-DplatformArtifactID` and `-DplatformVersion` configurable in the maven command ?

**Comment by spolti** [comment]
the MAVEN_APPEND_ARGS can't be used for it?

**Comment by spolti** [comment]
needs cherry-pick to main and 1.14.x branches.

File: modules/kogito-s2i-core/added/s2i-core [hunk 1]
```diff
@@ -98,18 +99,34 @@ function build_kogito_app() {
     if [ -f "pom.xml" ]; then
         log_info "---> Building application from source..."
         $MAVEN_HOME/bin/mvn clean package ${MAVEN_ARGS_APPEND} ${KOGITO_OPTS} ${nativeBuild} -s "${KOGITO_HOME}"/.m2/settings.xml \
-            -DskipTests -Dmaven.test.skip -Dmaven.javadoc.skip=true -Dmaven.site.skip=true -Dmaven.source.skip=true \
-            -Djacoco.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Dpmd.skip=true -Dfabric8.skip=true
+	    $(get_quarkus_platform_properties) -DskipTests -Dmaven.test.skip -Dmaven.javadoc.skip=true \
```
Comment (radtriste) [inline modules/kogito-s2i-core/added/s2i-core]: This is not needed here. Platform properties are only for the Quarkus Maven plugin

**Review by radtriste — CHANGES_REQUESTED** [review]

**Comment by radtriste** [comment]
jenkins retest this

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
@spolti could you please cherry-pick to `main` and `1.14.x` ?

**Pull request merged**